### DISCUSSION TITLE
Confusion about the effective active space hamiltonian h_eff

### DISCUSSION THREAD

**Discussion by tommontgomery450** [BoothGroup/Vayesta#46] [Q&A] [answered]
Hi, firstly thanks for building such a great library!

In the snippet below

```
# Calculate the effective onebody interaction within the cluster.
123 f_act = np.linalg.multi_dot((c_act.T, self.mf.get_fock(), c_act))
124 v_act = 2 * np.einsum('iipq->pq', eris[occ, occ]) - np.einsum('iqpi->pq', eris[occ, :, :, occ])
```

taken from 

dmet.fragment.DMETFragment.get_dmet_energy_contrib

 i'm trying to understand how `h_eff` is being calculated. My assumption (possibly incorrect) was that the summation indices, i, in the expression for v_act were over the inactive environment occupied orbitals but eris seems to be defined over just the active orbitals and taken from cluster_solver.get_eris(). The logic to get h_eff seems to be the same for the method on ClusterSolver and on the DumpSolver.

Any help understanding how the implementation relates to the equation for h_eff would be much appreciated.

Thanks

Tom

**Comment by tommontgomery450** [comment]
I may have answered my own question here...

The v_act here is to minus the contributions from occupied orbitals in the active space from the total coulomb and exchange contributions and thus leave just the total coulomb and exchange contributions from the inactive occupied orbitals.

i.e. since 

h_eff = f_act - v_act = h^core_act + sum{2g_iipq - g_iqpi}$ 

where i is over occupied inactive orbitals

Let me know if have made a mistake here!

**Comment by maxnus** [comment, marked as answer]
Hi Tom,

The summation `i` above is over the occupied cluster orbitals. This might be confusing but we build h_eff "backwards" - instead of adding the Coulomb and exchange interaction arising from the inactive occupied environment orbitals to the bare 1e-Hamiltonian, we remove the
the Coulomb and exchange interactions from active occupied cluster electrons from the **Fock** matrix. In this way we only require the ERIs in the pure cluster space (as you pointed out), but it will give the same result. Hope this helps.

PS. I see that you figured it out already!

**Reply by tommontgomery450** [reply]
Ah thanks for the quick feedback!
